src/Microsoft.ML.Dnn/LearningRateScheduler.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// A class that contains the current train state to use for learning rate scheduling.
+    /// </summary>
+    internal class TrainState
+    {
+        internal int CurrentBatchIndex;
+        internal int CurrentEpoch;
+        internal int BatchSize;
+        internal int BatchesPerEpoch;
+    }
+
+    /// <summary>
+    /// This interface defines a learning rate scheduler.
+    /// </summary>
+    public interface ILearningRateScheduler
```

Comment:
Instead of an interface, let's use an abstract class here. That way we can add more members to it in the future.